### DISCUSSION TITLE
How to get text inside of a Button centered?

### DISCUSSION THREAD

**Discussion by rainerschoe** [emilk/egui#2450] [Q&A]
Text in Button Widget is always left aligned when button has a `min_size`.
I tried to use LayoutJob like this:
```
            let mut job = egui::text::LayoutJob::default();
            //job.text = "asd".into();
            job.halign = egui::Align::CENTER;
            job.append(
                "Hello ",
                0.0,
                egui::TextFormat {
                    font_id: egui::FontId::new(14.0, egui::FontFamily::Proportional),
                    color: egui::Color32::WHITE,
                    ..Default::default()
                },
            );
            let eg_button = egui::Button::new(job).min_size(egui::Vec2{x:100.0, y:100.0});
            ui.add(eg_button);
```
but it lead to text being halfway out of the button.
`Align::LEFT` and `Align::RIGHT` also are not doing what I expect here.

Any help appreciated (I am new to egui :P )

**Comment by ahakkar** [comment]
Well I am also wondering this. There is no logical reason to NOT have the text centered in a button by default. The current left-justifying behaviour is extremely unexpected from standard UI elements.

E: you can horizontally center the text with something like

`ui.horizontal_centered(|ui| {
    if Self::button(ui) {
        should_close = true;
    }
});`

`fn button(ui: &mut Ui) -> bool {
    ui.add(
        Button::new(RichText::new("Close"))
            .min_size(constants::DEFAULT_BUTTON_SIZE),
    )
    .clicked()
}`
